crates/sui-sdk/examples/sign_tx_guide.rs

```diff
@@ -0,0 +1,149 @@
+use sui_types::signature::GenericSignature;
+use sui_types::{
+    base_types::SuiAddress,
+    crypto::{get_key_pair_from_rng, SuiKeyPair},
+};
+
+// This example walks through the Rust SDK usecase in sign-txn.mdx.
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // set up sui client for the desired network.
+    let sui_client = SuiClientBuilder::default().build_devnet().await?;
+
+    // deterministically generate a keypair, testing only, do not use for mainnet.
+    let skp_determ_0 =
+        SuiKeyPair::Ed25519(Ed25519KeyPair::generate(&mut StdRng::from_seed([0; 32])));
+    let _skp_determ_1 =
+        SuiKeyPair::Secp256k1(Secp256k1KeyPair::generate(&mut StdRng::from_seed([0; 32])));
+    let _skp_determ_2 =
+        SuiKeyPair::Secp256r1(Secp256r1KeyPair::generate(&mut StdRng::from_seed([0; 32])));
+
+    // randomly generate a keypair.
+    let _skp_rand_0 = SuiKeyPair::Ed25519(get_key_pair_from_rng(&mut rand::rngs::OsRng).1);
+    let _skp_rand_1 = SuiKeyPair::Secp256k1(get_key_pair_from_rng(&mut rand::rngs::OsRng).1);
+    let _skp_rand_2 = SuiKeyPair::Secp256r1(get_key_pair_from_rng(&mut rand::rngs::OsRng).1);
+
+    // import a keypair from a base64 encoded 32-byte `private key`.
+    let _skp_import_no_flag_0 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+    let _skp_import_no_flag_1 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+    let _skp_import_no_flag_2 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+
+    // import a keypair from a base64 encoded 33-byte `flag || private key`. The signature scheme is determined by the flag.
+    let _skp_import_with_flag_0 =
+        SuiKeyPair::decode_base64("ANRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_1 =
+        SuiKeyPair::decode_base64("AdRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_2 =
+        SuiKeyPair::decode_base64("AtRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+
+    // replace `skp_determ_0` with the variable names above
+    let pk = skp_determ_0.public();
+    let sender = SuiAddress::from(&pk);
+    println!("Sender: {:?}", sender);
+
+    // make sure the sender has a gas coin as an example.
+    let coin = sui_client
+        .coin_read_api()
+        .get_coins(sender, None, None, None)
+        .await?
+        .data
+        .into_iter()
+        .next();
+
+    if coin.is_none() {
+        println!("gas coin not found, refer to this re getting gas coin https://docs.sui.io/guides/developer/getting-started/get-coins");
+        return Ok(());
```

Comment:
Instead of returning, let's make it easier for the user by calling utils::request_tokens_from_faucet with the address that needs the coins.
```suggestion
        println!("Will request tokens from the faucet, please wait.");
        request_tokens_from_faucet(sender, &sui_client).await?;
```

Reply:
the call to request faucet hangs when i tried to call it at task id, not sure whats the right workaround

Reply:
Ok, I will have a look at it. It should simply wait until the coin is sent to the address.